src/report/errors.rs

```diff
@@ -27,6 +28,18 @@ use crate::report::{
 };
 use crate::token::{leak, Loc};
 
+/// Error types that should be logged once when consolidating reports
+static LOG_ONCE: LazyLock<TigerHashSet<ErrorKey>> = LazyLock::new(|| {
+    AHashSet::from([
```

Comment:
Using AHashSet here still bothers me. Is there some variation that will work, like `[A, B, C].into()` or `[A, B, C].iter().collect()`  ?